### ISSUE TITLE
Unable to get carousel working with dynamic card content created via fetching json

### ISSUE THREAD

**Issue by ezcg** [YIZHUANG/react-multi-carousel#155]
Hi,

Thanks for creating this carousel project.

In the carousel modification I am working on, instead of hard coding image urls in pages/index.js, I retrieve json via a separate component. However, I can't get the return value from the separate component to display properly. Everything gets nested inside the first div.

Here's the code I have set up: https://codeshare.io/5wzdrR

Heres' the resultant html: https://imgur.com/a/hgpuCCp

To summarize, instead of looping over an array of images inside of pages/index.js, I include a 'Cards' component which retrieves the json and builds the divs. But the divs all get put under a single bullet and only the first one gets displayed. 

Any ideas on how to get the returned divs to get their on 'li'? 

Thanks for any help.

**Comment by ezcg** [comment]
A solution that works is to use getInitialProps in pages/index.js. I can then iterate over the array in the Index class.

import fetch from 'isomorphic-unfetch'
Index.getInitialProps = async function() {
  const res = await fetch('http://abbotkinneybl.com/json/7.json');
  const data = await res.json();
  // appears as this.props.data in component
  return {data:Object.values(data)};
};
